src/elm/Session/LoggedIn.elm

```diff
@@ -74,6 +75,7 @@ init shared accountName flags =
     ( initModel shared authModel accountName flags.selectedCommunity
     , Cmd.batch
         [ Api.Graphql.query shared (Profile.query accountName) CompletedLoadProfile
+        , Api.Graphql.query shared (Community.communityQuery flags.selectedCommunity) CompletedLoadSettings
```

Comment:
Maybe we should do a query just to get the settings, to optimize our requests

Reply:
I disagree. We'd be doing extra work to save a few bytes on this request, all to be undone in case we need more context in the future. Doesn't sound like a worthy trade-off to me.

Reply:
@RodrigoLeiteF the problem appears on the `selectionSet` this function (`Community.communityQuery`) uses.

This is the definition:

```elm
communitySelectionSet : SelectionSet Model Cambiatus.Object.Community
communitySelectionSet =
    SelectionSet.succeed Model
        |> with Community.name
        |> with Community.description
        |> with (Eos.symbolSelectionSet Community.symbol)
        |> with Community.logo
        |> with (Eos.nameSelectionSet Community.creator)
        |> with Community.inviterReward
        |> with Community.invitedReward
        |> with Community.memberCount
        |> with (Community.members Profile.selectionSet)
        |> with (Community.objectives objectiveSelectionSet)
```

This makes we load members and objectives. Checkout the objective selection set.

```elm
objectiveSelectionSet : SelectionSet Objective Cambiatus.Object.Objective
objectiveSelectionSet =
    SelectionSet.succeed Objective
        |> with Objective.id
        |> with Objective.description
        |> with (Eos.nameSelectionSet Objective.creatorId)
        |> with (Objective.actions identity actionSelectionSet)
        |> with (Objective.community communitiesSelectionSet)
```

Then we load actions inside of it (also we reload community data **again** via `communitiesSelectionSet`)

```elm

actionSelectionSet : SelectionSet Action Cambiatus.Object.Action
actionSelectionSet =
    SelectionSet.succeed Action
        |> with Action.id
        |> with Action.description
        |> with Action.reward
        |> with Action.verifierReward
        |> with (Eos.nameSelectionSet Action.creatorId)
        |> with (Action.validators Profile.selectionSet)
        |> with Action.usages
        |> with Action.usagesLeft
        |> with Action.deadline
        |> with Action.verificationType
        |> with Action.verifications
        |> with Action.isCompleted

```

Here we also load the profile of all the validators for all the actions. (Muda usually puts 8+)

So the trade off is:

1) Keep the query as it is, loading this bunch of stuff I've just mentioned. No further work is needed
2) We create a new selection set and type to query only the required data, not asking for more. More work needed but it is little, one type, one function

After this context, do you still think its not worthy? If so can you explain your line of thought?

Reply:
just to give an actual example of my thought. This is the generated query from that function `Community.communityQuery`

```graphql
query {
	community2197679141: community(symbol: "MUDA") {
		objectives {
			community {
				memberCount1207450440: memberCount
				creator3832528868: creator
				logo3832528868: logo
				symbol3832528868: symbol
				description3832528868: description
				name3832528868: name
			}
			actions {
				isCompleted3880003826: isCompleted
				verifications1207450440: verifications
				verificationType1066107094: verificationType
				deadline3432121944: deadline
				usagesLeft1207450440: usagesLeft
				usages1207450440: usages
				validators {
					analysisCount1207450440: analysisCount
					communities {
						logo3832528868: logo
						name3832528868: name
						symbol3832528868: symbol
					}
					chatToken12867311: chatToken
					chatUserId12867311: chatUserId
					interests12867311: interests
					avatar12867311: avatar
					account3832528868: account
					location12867311: location
					bio12867311: bio
					email12867311: email
					name12867311: name
				}
				creatorId3832528868: creatorId
				verifierReward3770192905: verifierReward
				reward3770192905: reward
				description3832528868: description
				id1207450440: id
			}
			creatorId3832528868: creatorId
			description3832528868: description
			id1207450440: id
		}
		members {
			analysisCount1207450440: analysisCount
			communities {
				logo3832528868: logo
				name3832528868: name
				symbol3832528868: symbol
			}
			chatToken12867311: chatToken
			chatUserId12867311: chatUserId
			interests12867311: interests
			avatar12867311: avatar
			account3832528868: account
			location12867311: location
			bio12867311: bio
			email12867311: email
			name12867311: name
		}
		memberCount1207450440: memberCount
		invitedReward3770192905: invitedReward
		inviterReward3770192905: inviterReward
		creator3832528868: creator
		logo3832528868: logo
		symbol3832528868: symbol
		description3832528868: description
		name3832528868: name
	}
}
```

This is the result:

(note: I've omitted email, name, bio and location to keep users privacy)

**JSON was too large to paste here, I'll be sending it over our slack**

Reply:
With further context, I agree. I didn't realize we pulled so much data on the community query. Makes me wonder if there are other places where we can pull less data as well :) Thanks for the insight, I'll make the change.

Reply:
Done!